addons/equipviewer/icon_extractor.lua

```diff
@@ -0,0 +1,255 @@
+        dat_stats.file = icon_file
+    end
+    return icon_file
+end
+
+-- 32 bit color palette-indexed bitmaps. Bits are rotated and must be decoded.
+local encoded_to_decoded_char = {}
+local encoded_byte_to_rgba = {}
+local alpha_encoded_to_decoded_adjusted_char = {}
+local decoded_byte_to_encoded_char = {}
+for i = 0, 255 do
+    encoded_byte_to_rgba[i] = ''
+    local n = (i % 32) * 8 + floor(i / 32)
+    encoded_to_decoded_char[char(i)] = char(n)
+    decoded_byte_to_encoded_char[n] = char(i)
+    n = n * 2
+    n = n < 256 and n or 255
+    alpha_encoded_to_decoded_adjusted_char[char(i)] = char(n)
+end
+local decoder = function(a, b, c, d)
+    return encoded_to_decoded_char[a]..
+        encoded_to_decoded_char[b]..
+        encoded_to_decoded_char[c]..
+        alpha_encoded_to_decoded_adjusted_char[d]
+end
+function convert_item_icon_to_bmp(data)
+    local color_palette = string.gsub(sub(data, 1, 1024), '(.)(.)(.)(.)', decoder)
+    -- rather than decoding all 2048 bytes, decode only the palette and index it by encoded byte
+    for i = 0, 255 do
+        local offset = i * 4 + 1
+        encoded_byte_to_rgba[decoded_byte_to_encoded_char[i]] = sub(color_palette, offset, offset + 3)
+    end
+
+    return header .. string.gsub(sub(data, 1025, 2048), '(.)', function(a) return encoded_byte_to_rgba[a] end)
+end
+
+
+local buff_dat_map = {
+    [1]={min=0, max=1024, dat_path='119/57', offset=0},
+}
+function find_buff_dat_map(id)
+    for _,stats in pairs(buff_dat_map) do
+        if id >= stats.min and id <= stats.max then
+            return stats
+        end
+    end
+    return nil
+end
+local buff_by_id = function (id, output_path)
+    local dat_stats = find_buff_dat_map(id)
+    local icon_file = open_dat(dat_stats)
+    
+    local id_offset = dat_stats.min + dat_stats.offset
+    icon_file:seek('set', (id - id_offset) * 0x1800)
+    local data = icon_file:read(0x1800)
+
+    bmp = convert_buff_icon_to_bmp(data)
+
+    local f = io.open(output_path, 'wb')
+    f:write(bmp)
+    coroutine.yield()
+    f:close()
+end
+icon_extractor.buff_by_id = buff_by_id
+
+
+local ffxi_path = function(location)
+    game_path = location
+end
+icon_extractor.ffxi_path = ffxi_path
+
+
+-- A mix of 32 bit color uncompressed and *color palette-indexed bitmaps
+-- Offsets defined specifically for status icons
+-- * some maps use this format as well, but at 512 x 512
+function convert_buff_icon_to_bmp(data)
+    local length = byte(data, 0x282) -- The length is technically sub(0x281, 0x284), but only 0x282 is unique
+
+    if length == 16 then -- uncompressed
+        data = sub(data, 0x2BE, 0x12BD)
+        data = string.gsub(data, '(...)\128', '%1\255') -- All of the alpha bytes are currently 0 or 0x80.
+    elseif length == 08 then -- color table
+        local color_palette = sub(data, 0x2BE, 0x6BD)
+        color_palette = string.gsub(color_palette, '(...)\128', '%1\255')
+    
+        local n = 0
+        for i = 1, 1024, 4 do
```

Comment:
Another very minor thing, but same as before, `0x400` would fit with the above, since you do steps of 4, and `4 * 0x400 == 0xC00`, so the numbers all line up :)

Reply:
Would you like me to just change most of the numbers to hex? Or just these two?